addon/base-record-array.js

```diff
@@ -199,6 +197,23 @@ if (CUSTOM_MODEL_CLASS) {
       });
     }
 
+    // Need to subclass `removeAt`, `pushObject`, and `insertAt` because the default implementations by
+    // MutableArray will end up calling replaceInNativeArray and not our own replace after an `isArray` check
+    // https://github.com/emberjs/ember.js/blob/21bd70c773dcc4bfe4883d7943e8a68d203b5bad/packages/%40ember/-internals/metal/lib/array.ts#L27
+    // https://github.com/emberjs/ember.js/blob/21bd70c773dcc4bfe4883d7943e8a68d203b5bad/packages/%40ember/-internals/metal/lib/array.ts#L38
+    removeAt(index, len = 1) {
+      this.replace(index, len, []);
+      return this;
+    }
+
+    pushObject(obj) {
+      return this.insertAt(this.length, obj);
+    }
+
+    insertAt(idx, object) {
+      return this.replace(idx, 0, [object]);
+    }
```

Comment:
shouldn't these all be deprecated as well? (i.e. non-native methods)?

Reply:
Given how layered these methods are, it seems better to deprecate once in `replace` and `objectAt` rather than try to hit each method individually. I'll do that separately from this PR

Reply:
Does it make sense to deprecate the non-native methods? If you were using a JS array, you would be using the ember array methods for mutating it if you wanted the changes to be observable. If we deprecate them, it becomes harder to write functions that work on both JS arrays and M3 Arrays.